### PR TITLE
Week2 - Implementation of Daily and Week Assignments in NodeJS 

### PR DESCRIPTION
**Pull request by aditya-s-19**
(1) Briefly describe your change.
Added 2 assignments (2 daily )
However they encorporate all the assignments in it
The second assignment uses JWT tokens for authentication with middleware

(2) What parts of the application are affected by your change?
None

(3) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Yes. I checked with Postman app to test my api

### CODE REVIEW

**Review by RitikaGarg-ITT — APPROVED** [review]

**Review by choudhary-keshav — APPROVED** [review]

File: Week-2/Daily-Assignments/Assignment2/assignment2.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const bearerHeader = req.headers["authorization"];
+  if (bearerHeader) {
+    const bearer = bearerHeader.split(" ");
+    const token = bearer[1];
+    return token;
+  }
+  return "";
+};
+
+const auth = (req: Request, res: Response, next: NextFunction): void => {
```
Comment (shreyansh-khatri) [inline Week-2/Daily-Assignments/Assignment2/assignment2.ts]: please use full name for this function

File: Week-2/Daily-Assignments/Assignment2/assignment2.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  const token = getToken(req);
+  if (!token) {
+    return res.send("Invalid token! Access not allowed");
```
Comment (shreyansh-khatri) [inline Week-2/Daily-Assignments/Assignment2/assignment2.ts]: try to use status codes also in all your response

**Review by shreyansh-khatri — CHANGES_REQUESTED** [review]

**Review by shreyansh-khatri — APPROVED** [review]

File: Week-2/Daily-Assignments/Assignment2/tempCodeRunnerFile.ts [hunk 3]
```diff
@@ -0,0 +1 @@
+ JSON.parse(response)
```
Comment (anusreesubash) [inline Week-2/Daily-Assignments/Assignment2/tempCodeRunnerFile.ts]: where is this file used?

Comment (aditya-s-19) [reply Week-2/Daily-Assignments/Assignment2/tempCodeRunnerFile.ts]: it was accidentally created by code runner extension. ill remove it

**Review by Harrrrry — APPROVED** [review]

**Pull request merged**